### PR TITLE
Server fixes to support running against QEMU/libvfio-user

### PR DESCRIPTION
**Pull request by rbradford**
- server: Fix negotiation support
- server: Respond with EINVAL error if region out of range
- server: Generate error response on IRQ out of range
- server: Use zero minor version number
- server: Support DMA_MAP command with no FDs
- server: REGION_READ response size should contain payload size
- server: Add missing field to DeviceGetInfo
- server: Respond to DMA_UNMAP message with correct size
- server: Fill argsz in correctly for DEVICE_GET_INFO
- gpio: Handle larger access to PCI config region
- gpio: Import pci_configuration.rs from crosvm

### CODE REVIEW

File: examples/gpio/main.rs [hunk 1]
```diff
@@ -83,13 +83,35 @@ impl ServerBackend for TestBackend {
...
+                    reg_idx += 1;
+                    data_offset += 4;
+                }
+            } else {
+                // Require access to be naturally aligned
+                let reg_idx = offset as usize / 4;
+                let reg_offset = offset as usize % 4;
+                match reg_offset {
+                    0 => {}
+                    1 | 3 => assert!(len == 1),
```
Comment (sboeuf) [inline examples/gpio/main.rs:108]: For the case `reg_offset` is `1`, can't the `len` be 1, 2 or 3?

**Review by sboeuf — APPROVED** [review]
Nice PR :)

File: examples/gpio/main.rs (hunk 1 above)
Comment (rbradford) [reply examples/gpio/main.rs:108]: So the goal of these assertions are to identify a programming error - and in this case that error is making non naturally aligned PCI config space accesses. The only legitimate access size for offsets of 1 or 3 is a single byte. For offsets of 2 would be a byte or word. For an offset of 0 would be byte, word or dword.

Comment (sboeuf) [reply examples/gpio/main.rs:108]: Gotcha!

**Pull request merged**